storage/kubernetes/client.go

```diff
@@ -333,7 +333,10 @@ func newClient(cluster k8sapi.Cluster, user k8sapi.AuthInfo, namespace string, l
 
 	logger.Infof("kubernetes client apiVersion = %s", apiVersion)
 	return &client{
-		client:     &http.Client{Transport: t},
+		client: &http.Client{
+			Transport: t,
+			Timeout:   30 * time.Second,
```

Comment:
can we make this shorter? e.g. 15 seconds. There's no reason any of these requests should take 30 seconds.